python/pyarrow/tests/test_table.py

```diff
@@ -143,14 +140,15 @@ def test_chunked_array_to_numpy():
 
 
 def test_chunked_array_mismatch_types():
-    with pytest.raises(TypeError):
+    msg = "chunks must all be same type"
+    with pytest.raises(pa.ArrowInvalid, match=msg):
```

Comment:
Hmm... interesting. I would be in favor of changing the C++ side to return `TypeError` instead. Would you like to do that in this PR?